liblangutil/SourceReferenceFormatter.cpp

```diff
@@ -164,9 +174,19 @@ void SourceReferenceFormatter::printSourceLocation(SourceReference const& _ref)
 void SourceReferenceFormatter::printExceptionInformation(SourceReferenceExtractor::Message const& _msg)
 {
 	// exception header line
-	errorColored() << _msg.severity;
+	errorColored() << _msg.severity_error;
+    warningColored() << _msg.severity_warning;
+    infoColored() << _msg.severity_info;
+
 	if (m_withErrorIds && _msg.errorId.has_value())
 		errorColored() << " (" << _msg.errorId.value().error << ")";
+
+    else if(m_withWarningIds && _msg.warningId.has_value())
+        warningColored() << " (" << _msg.warningId().error << ")";
+
+    else if(m_withInfoIds && _msg.infoId.has_value())
+        warningColored() << " (" << _msg.infoId().error << ")";
```

Comment:
There are no members called `m_withWarningIds` and `m_withInfoIds`. Just like there is no `warningId()` and `infoId()` in `SourceReferenceExtractor::Message`.

All messages are associated with an **error id** regardless of their severity. I.e. it's still called error ID, even for warnings and infos.

Reply:
Fixed